src/App.js

```diff
@@ -127,32 +127,14 @@ const render = ({
           </Switch>
           <Modal
             isOpen={!!modalState.component}
-            style={{
-              overlay: {
-                position: 'fixed',
-                top: '0px',
-                left: '0px',
-                right: '0px',
-                bottom: '0px',
-                backgroundColor: 'rgba(0, 0, 0, 0.5)',
-                display: 'block',
-                zIndex: '111',
-              },
-              content: {
-                position: 'initial',
-                border: '1px solid rgb(204, 204, 204)',
-                background: 'rgb(255, 255, 255)',
-                borderRadius: '4px',
-                margin: '30px auto',
-                width: '55%',
-                boxShadow: 'rgba(0, 0, 0, 0.5) 0px 5px 15px',
-                overflow: 'visible',
-              },
+            {...{
+              unsetModal: effects.unsetModal,
```

Comment:
Probably makes sense to make modal get all these arguments from the provider itself. so this could end up being just `<Modal />`

Reply:
oh but other components need to call the `setModal` and `unsetModal`, so don't we need to provide the state from the top level?

Reply:
oops, I should have been clearer. 
You would leave the provider where it is. But just add `injectState` to the modal. Then it will get all those values as well.

Reply:
oh yeah ok now I see how this works :)))